databuilder/sqlalchemy_utils.py

```diff
@@ -190,3 +193,42 @@ def fetch_table_in_batches(
             break
         else:
             min_key = row[key_column_index]
+
+
+class ExecuteWithRetry:
```

Comment:
I think it's mostly because earlier iterations of the design needed a class and then I managed to simplify it but left the class in place because I didn't think it would massively simplify things to do it otherwise. It would be easy enough to make it a closure, `execute_with_retry_factory()` or something. It doesn't seem obviously more readable to me, but I'd be very happy to change it if others would.

Reply:
I tried out the closure version and it did look simpler overall, so I'm using that instead. Thanks for the prompt!

Reply:
It looks simpler to me too 🙂